deployment/ccip/changeset/v1_5_1/cs_add_token_e2e.go

```diff
@@ -39,96 +40,106 @@ import (
-//
-//     i. it adds chain support with the desired rate limits
-//     i. it adds the desired remote pool addresses to the token pool on the chain
-//     iii. if there used to be an existing token pool on tokenadmin_registry, it adds the remote pool addresses of that token pool to ensure 0 downtime
+// 3. Proposes admin rights for the token on the token admin registry
 //
-// 4. Proposes admin rights for the token on the token admin registry
+//		If the token admin is not an external address -
+//		3a. Accepts admin rights for the token on the token admin registry
+//		3b. Sets the pool for the token on the token admin registry
 //
-// If the token admin is not an external address -
-// 5. Accepts admin rights for the token on the token admin registry
-// 6. Sets the pool for the token on the token admin registry
+//	 4. Configures pools -
+//		   The configuration is set by via the input field ConfigurePools. This is so that we can have flexibility during configuration.
+//		   For ex: deploy chainB token, pool and configure with already deployed chainA tokenPool.
+//		   If config is not provided in the input, it skips this step.
 var AddTokensE2E = cldf.CreateChangeSet(addTokenE2ELogic, addTokenE2EPreconditionValidation)
 
 type E2ETokenAndPoolConfig struct {
-	TokenDeploymentConfig *DeployTokenConfig    // TokenDeploymentConfig is optional. If provided, it will be used to deploy the token and populate the pool deployment configuration.
-	DeployPoolConfig      *DeployTokenPoolInput // Deployment configuration for pools is not needed if tokenDeploymentConfig is provided. This will be populated from the tokenDeploymentConfig if it is provided.
-	ExistingPoolType      *cldf.ContractType    // Provide if pool is already deployed and you want to just configure it
-	PoolVersion           semver.Version
-	ExternalAdmin         common.Address // ExternalAdmin is the external administrator of the token pool on the registry.
-	RateLimiterConfig     RateLimiterPerChain
+	// TokenDeploymentConfig is optional. If provided, it will be used to deploy the token
+	// and populate the pool deployment configuration.
+	TokenDeploymentConfig *DeployTokenConfig `json:"tokenDeploymentConfig,omitempty"`
+
+	// Deployment configuration for pools is not needed if tokenDeploymentConfig is provided.
+	// This will be populated from the tokenDeploymentConfig if it is provided.
+	DeployPoolConfig *DeployTokenPoolInput `json:"deployPoolConfig,omitempty"`
+
+	// Provide if pool is already deployed and you want to just configure it
+	ExistingPoolType *cldf.ContractType `json:"existingPoolType,omitempty"`
+
+	// Version of the pool being deployed.
+	PoolVersion semver.Version `json:"poolVersion"`
+
+	// ExternalAdmin is the external administrator of the token pool on the registry.
+	ExternalAdmin common.Address `json:"externalAdmin"`
+
+	// Configuration for the rate limiter per chain.
+	RateLimiterConfig RateLimiterPerChain `json:"rateLimiterConfig"`
```

Comment:
Is it needed separately if ConfigurePool is directly taken from user?

Reply:
addressed here: https://github.com/smartcontractkit/chainlink/pull/17792/commits/08d2a9719a71db1808ec9f881f8e44b67fafe182